packages/fxa-email-service/src/db/core/mod.rs

```diff
@@ -33,18 +36,20 @@ use crate::{settings::Settings, types::error::AppResult};
 /// as a safeguard against leaking PII.
 #[derive(Debug)]
 pub struct Client {
-    client: RedisClient,
+    client: Mutex<RedisClient>,
```

Comment:
Methods on `redis::Client` are now `&mut self`, which means we either need to bubble `&mut self` up through our own callstack, or add interior mutability. I opted for the latter as a lower-impact change.

`Cell` was not an option though, because this struct is passed around as Rocket-managed state so it has to be threadsafe. There's no benefit to `RwLock` over `Mutex` in this case because all the usage requires `mut`. Ergo `Mutex`.

Reply:
Actually, I changed my mind while typing that comment. Bubbling the mutability seems simpler than using a mutex, I'll have a look at it...

Reply:
> Bubbling the mutability seems simpler than using a mutex, I'll have a look at it...

It doesn't buy us anything actually, because this is all stuff that gets passed around as `rocket::State<...>`. So it would be `rocket::State<Mutex<...>>` instead and we'd just be pushing the problem up to a higher level.